beacon_node/lighthouse_network/gossipsub/src/behaviour.rs

```diff
@@ -2706,8 +2712,8 @@ where
 
         for peer_id in recipient_peers {
             let Some(peer) = self.connected_peers.get_mut(peer_id) else {
-                tracing::error!(peer = %peer_id,
```

Comment:
instead of removing this error, can we instead filter above for the gossip promises? I.e.
```rust
// Gossip promises are kept for disconnected peers.
let iwant_peers = self
            .gossip_promises
            .peers_for_message(msg_id)
            .iter()
            .filter(|peer_id| self.connected_peers.contains_key(peer_id));
```

Reply:
I was thinking about this. I don't mind. 
The logic that went through my head was:
Adding this check adds a hashmap lookup for each peer, but a few lines down the page we do the exact same check.

I figured we are already doing the check, by just removing the error, we have no extra hash lookup, however the cost is we don't log an error in case a mesh peer isn't in the connected mapping (but we have these errors elsewhere). 

Its such a tiny difference, so happy to do whatever you think is best, just explaining why I removed the error.

Reply:
thanks for elaborating age! Do you know where else we check? It def doesn't make sense to double check this adding another iteration as you refer, I just think we shouldn't stop logging for this event

Reply:
So yeah happy to do what you prefer. 

I can't think of a way to keep the error without adding an extra hashmap lookup. Its a trivial lookup, and maybe the error is worth it. So I'm happy to go with the soln you have suggested here if you want the error.